velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -359,26 +393,85 @@ bool registerAverage(const std::string& name) {
         if (exec::isRawInput(step)) {
           switch (inputType->kind()) {
             case TypeKind::SMALLINT:
-              return std::make_unique<
-                  AverageAggregate<int16_t, double, double>>(resultType);
+              if (aggregateType == AggregateType::kAvg) {
+                return std::make_unique<AverageAggregate<
+                    int16_t,
+                    double,
+                    double,
+                    AggregateType::kAvg>>(resultType);
```

Comment:
No, it won't allow variables in the template instantiation

Reply:
Maybe then abstract a function (which is templatized with all the types as parameters) and which does the conditional in its body so that we avoid similar logic for each type.